src/Compilers/Core/Portable/SourceGeneration/AbstractSourceGenerator.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.SourceGeneration
+{
+    /// <summary>
+    /// A base class that can be used to implement a Source Generator 
+    /// </summary>
+    public abstract class SourceGenerator : ISourceGenerator
```

Comment:
Is there a risk that the existence of both an interface and a simple abstract class will lead to confusion over which to implement? It might be simpler overall to have just the interface.

Reply:
For all the other similar features, we only have an abstract class.

Reply:
Yeah, agree it's slightly odd, but we explicitly wanted to make this an interface based implementation so we could support the scenario of Analyzer/Generator combos.

The issue we've seen with just the interface is you often don't need the init method, which means even the simplest of generators has to have an empty method. We could always split it out into two interfaces? `ISourceGenerator` and `ISourceGeneratorWithInit : ISourceGenerator` ?

Reply:
> so we could support the scenario of Analyzer/Generator combos.

💭 It's a bit of a strange thing to support, since they have completely different execution characteristics. If one did come through review like that, I would probably request it be split up anyway for clarity.